src/org/sosy_lab/java_smt/solvers/z3/Z3Model.java

```diff
@@ -375,9 +378,14 @@ public void close() {
   }
 
   @Override
-  protected Long evalImpl(Long formula) {
+  protected Long evalImpl(Long formula) throws InterruptedException {
     LongPtr resultPtr = new LongPtr();
-    boolean satisfiableModel = Native.modelEval(z3context, model, formula, false, resultPtr);
+    boolean satisfiableModel;
+    try {
+      satisfiableModel = Native.modelEval(z3context, model, formula, false, resultPtr);
+    } catch (Z3Exception e) {
+      throw new InterruptedException("Z3 model evaluation was interrupted.");
+    }
```

Comment:
This code is wrong. Z3Exceptions can have other causes as well. Please use proper handling from `Z3FormulaCreator.handleZ3Exception(e)`.

Reply:
Thanks for noticing! You already changed it, thanks!